script.js

```diff
@@ -1,4 +1,107 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+//game mode
+var diceRollMode = "dice rolling!";
+var diceOrderMode = "choose the order of dice";
+var beatThatWinner = "comparing the numbers";
+//default game mode
+var gameMode = diceRollMode;
+// players
+var player1 = "Player 1";
+var player2 = "Player 2";
+var activePlayer = player1;
+// arrays to store dice rolls
+var player1Dice = [];
+var player2Dice = [];
+// player dice order
+var player1Order;
+var player2Order;
+//win counter
+var player1Win = 0;
+var player2Win = 0;
+
+// random Dice Roll function
+var diceRoll = function () {
+  var diceRoll = Math.floor(Math.random() * 6) + 1;
+  return diceRoll;
+};
+
+// storing dice into arrays
+var dicePush = function () {
+  var dicePush = [diceRoll(), diceRoll()];
+
+  // Checking if player is 1 or 2, then pushing the dice numbers into the array
+  if (activePlayer == player1 && gameMode == diceRollMode) {
+    player1Dice = dicePush;
+  } else if (activePlayer == player2) {
+    player2Dice = dicePush;
+  }
+  return dicePush;
```

Comment:
We usually don't assign arrays like this. It's better to do the push method
Reason being this idea of reference vs primitive types. - I'll run through this in class !

[Watch this video to find out more :)](https://www.youtube.com/watch?v=9ooYYRLdg_g)